src/pkg/util/fs/mount/mount.go

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package mount
+
+import (
+	"fmt"
+	"strings"
+	"syscall"
+
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+var mountFlags = []struct {
+	option string
+	flag   uintptr
+}{
+	{"acl", 0},
+	{"async", 0},
+	{"atime", 0},
+	{"bind", syscall.MS_BIND},
+	{"defaults", 0},
+	{"dev", 0},
+	{"diratime", 0},
+	{"dirsync", 0},
+	{"exec", 0},
+	{"iversion", 0},
+	{"lazytime", 0},
+	{"loud", 0},
+	{"mand", 0},
+	{"noacl", 0},
+	{"noatime", 0},
+	{"nodev", syscall.MS_NODEV},
+	{"nodiratime", 0},
+	{"noexec", syscall.MS_NOEXEC},
+	{"noiversion", 0},
+	{"nolazytime", 0},
+	{"nomand", 0},
+	{"norelatime", 0},
+	{"nostrictatime", 0},
+	{"nosuid", syscall.MS_NOSUID},
+	{"rbind", syscall.MS_BIND | syscall.MS_REC},
+	{"relatime", 0},
+	{"remount", syscall.MS_REMOUNT},
+	{"ro", syscall.MS_RDONLY},
+	{"rw", 0},
+	{"silent", syscall.MS_SILENT},
+	{"strictatime", 0},
+	{"suid", 0},
+	{"sync", 0},
+}
+
+type fsContext struct {
+	context bool
+}
+
+var authorizedImage = map[string]fsContext{
+	"ext3":     {true},
+	"squashfs": {true},
+}
+
+var authorizedFS = map[string]fsContext{
+	"overlay": {true},
+	"tmpfs":   {true},
+	"ramfs":   {true},
+	"devpts":  {true},
+	"sysfs":   {false},
+	"proc":    {false},
+	"mqueue":  {false},
+}
+
+// ConvertOptions converts an options string into a pair of mount flags and mount options
+func ConvertOptions(options []string) (uintptr, []string) {
+	var flags uintptr
+	finalOptions := []string{}
+	isFlag := false
+
+	for _, option := range options {
+		optionTrim := strings.TrimSpace(option)
+		for _, flag := range mountFlags {
+			if flag.option == optionTrim {
+				flags |= flag.flag
+				isFlag = true
+				break
+			}
+		}
+		if !isFlag {
+			finalOptions = append(finalOptions, optionTrim)
+		}
+		isFlag = false
+	}
+	return flags, finalOptions
+}
+
+// Points defines and stores a set of mount points
+type Points struct {
+	Context string
```

Comment:
Would it be better to have points be like this:

```
type Point struct {
    Context string
    point specs.Mount
}

var Points []Point
```

This way each point can have unique context? Or maybe I don't quite understand the use of `Context string` yet and it's really necessary only for the entire set of `Points`?

Reply:
Context is for SELinux mount context, OCI allow to defines one mountlabel, I understood that as one mountlabel for all mount points. That's the meaning of Context here

Reply:
Ok cool, in which case makes sense to leave this as is